### DISCUSSION TITLE
[Ente Photos] background loading of videos

### DISCUSSION THREAD

**Discussion by wunderox** [ente-io/ente#1306] [General]
After a brief discussion with Vishnu himself on Matrix, and at his request, I'm opening a discussion based on a detail and specific aspect of ente Photos desktop.

Currently, in Ente Photos for linux (1.6.63, in RPM) and maybe other desktop versions too, when trying to play a video already uploaded, a loading circle starts in the upper right corner to indicate the downloading/loading/buffering of the video. If the user decides to close the video for whatever reason, before the video finishes loading,  the video will continue loading in the background.

This leads to two questions:
1) Is this the expected behavior of users? The reason this is currently done is such that the user does not need to rebuffer the video from scratch if the user decides to open the video again. Do some users expect the video loading/buffering to stop when the video is closed?
2) In the case where this is the expected behavior, should the desktop application somehow inform the user that this is the case? If yes, how should it do so? Should it use something like the loading widget used for when uploading content to Ente Photos? Should it be cancellable? Or should it be something else?

We are here trying to collect more perspectives from users, to first know how they expected ente Photos desktop to behave in this situation, and with the information here, to know what their opinion is on the topic.

Thank you all.

Edit: small clarifications.

**Comment by mxmehl** [comment]
I noticed this as well, thanks for opening the discussion.

For me, this is a rather unwanted behaviour. Sometimes I search for a specific video and have to open one or more to see whether they match, which I often cannot just from the thumb image.

But when I notice that a video is not of interest for me, it's still fully downloaded, using up a lot of bandwidth/traffic. So it would be great if the app stopped downloading the video when I stop watching the video.

However, when the video is already fully downloaded, I would of course love to keep it cached, but AFAIU this already is the case.

**Reply by vishnukvmd** [reply]
Thanks for sharing your perspective! And yes, downloaded items are stored in cache.
